src/main/kotlin/no/nav/yrkesskade/ysmeldingapi/services/SkademeldingService.kt

```diff
@@ -124,15 +134,14 @@ class SkademeldingService(private val skademeldingInnsendingClient: Skademelding
             Tidstype.periode -> validerTidsperiode(skademelding.hendelsesfakta.tid.periode)
         }
 
-        // Hent kodelister basert på rolletype som skal benyttes for å finne gyldige verdier
-        check(skademelding.skadelidt!!.dekningsforhold.rolletype != null, { "rolletype er påkrevd" })
-
-        val rolletype = skademelding.skadelidt!!.dekningsforhold.rolletype
-
-        kodeverkValidator.sjekkGyldigKodeverkverdiForType(rolletype, "rolletype", "${rolletype} er ikke en gyldig rolletype kode i kodelisten")
+        // dersom tidstype er periode - valider sykdomsfelter
+        if (skademelding.hendelsesfakta.tid.tidstype == Tidstype.periode) {
+            kodeverkValidator.sjekkGyldigKodeverkverdiForType(skademelding.skade.sykdomstype.orEmpty(), "sykdomstype", "${skademelding.skade.sykdomstype} er ikke en gyldig verdi")
```

Comment:
Vil orEmpty() ta hensyn til at det kan være valgt tidstype periode, men at det ikke er yrkessykdom? Bra!!

Reply:
Denne er fjernet. Sykdomstype er en del av kodeverket for skadetDel skadeart.